Drive Documentation

jeffaburke opened the pull request:
Closes #10

jeffaburke committed:
Documentation

jeffaburke committed:
Documentation

msoucy commented on src/main/java/frc/robot/subsystems/Drive.java:
```diff
@@ -10,19 +10,50 @@
+ * 
+ * 5) Does it store any state? Right and Left motor groups.
+ * 
+ * 6) Sensors? Speed Controllers
```
These aren't sensors

msoucy replied:
Are there encoders? Gyros?

msoucy commented on src/main/java/frc/robot/subsystems/Drive.java:
```diff
@@ -10,19 +10,50 @@
+ * 1) What does it do? Makes motors turn a certain amount depending on how much
+ * an axis is triggered
+ * 
+ * 2) What modes does it have? Has a mode to differentiate between right and
```
This is not a mode - those are just members.

msoucy commented on src/main/java/frc/robot/subsystems/Drive.java:
```diff
@@ -10,19 +10,50 @@
+ * 3) What interactions does it have with other subsystems? Does not have any
+ * interactions with other systems other than the space it shares on the robot
+ * 
+ * 4) How is it triggered -> OI? Triggered by motors to make the robots wheels
```
Triggered **by** motors?

jeffaburke committed:
added a differential drive maps

jeffaburke committed:
added a differential drive maps

jeffaburke committed:
Fixes for comments by MSoucy and Ben

jeffaburke committed:
Fixes for comments by MSoucy and Ben

DavidDesfosse committed:
Added driveFlip for Sean

Now you can hit a button and invert the drive direction

DavidDesfosse committed:
Added driveFlip for Sean

Now you can hit a button and invert the drive direction

bot190 commented on src/main/java/frc/robot/subsystems/Drive.java:
```diff
@@ -2,33 +2,69 @@
+     * Maps the drive axises
+     * 
+     * @param forward defines the forward direction
+     * @param turn    defines the direction to turn
```
These don't really "define" the directions, they provide the amount of forward/turning speed.

bot190 commented on src/main/java/frc/robot/subsystems/Drive.java:
```diff
@@ -2,33 +2,69 @@
         }, this);
     }
+
+    public CommandBase driveFlip(DoubleSupplier forward, DoubleSupplier turn) {
```
Maybe `reversedDrive`
This should have the same comment structure as the `drive` command. 

It should also be bound to a button with `toggleWhenActive`.

bot190 commented on src/main/java/frc/robot/maps/TempestMap.java:
```diff
@@ -53,4 +46,10 @@ public LiftMap getLiftMap() {
     }
+
+    @Override
+    public IntakeMap getIntakeMap() {
```
This shouldn't be here.

bot190 commented on src/main/java/frc/robot/subsystems/Drive.java:
```diff
@@ -2,33 +2,69 @@
+
+    public CommandBase driveFlip(DoubleSupplier forward, DoubleSupplier turn) {
+        return new RunCommand(() -> {
+            double yAxis = forward.getAsDouble();
```
This should really just be providing a new DoubleSupplier to `drive` that reverses the direction.

bot190 reviewed (changes requested):
A few comments to cleanup the `driveFlip` command

tpukg committed:
Removed Intake Map from TempestMap and made changes to driveFlip

tpukg committed:
I put back the auto generated method

tpukg committed:
Added reverse function and mapped it

bot190 commented on src/main/java/frc/robot/subsystems/Intake.java:
```diff
@@ -45,4 +45,4 @@ public StartEndCommand discharge() {
 
-}
\ No newline at end of file
+    }
```
This change shouldn't be here.

msoucy replied:
Please undo this

bot190 commented on src/main/java/frc/robot/subsystems/Drive.java:
```diff
@@ -2,32 +2,60 @@
 
-    public Drive(RobotMap.DriveMap map) {
+    /**
+     * Defines the left and right motor(s) and then puts them into a differential
```
This doesn't `Define` motors, it gets them from the robot map. Please update this comment.

bot190 commented on src/main/java/frc/robot/maps/RobotMap.java:
```diff
@@ -1,23 +1,15 @@
-    public DriveMap getDriveMap() {
-        return new DriveMap();
+    public DifferentialDriveMap getDriveMap() {
+        return null;
```
This seems dangerous. We should provide a mock drive map, probably in ChopShopLib. For now this should provide mock speed controllers

bot190 reviewed (changes requested):
A few more changes.

DavidDesfosse committed:
Fixed Ben's comments

Changed a comment to have correct information and made the differential drive map get mock speed controllers

DavidDesfosse committed:
Fixed merge conflicts involving the drive changes

I hate merge conflicts.

msoucy commented on src/main/java/frc/robot/maps/HyperionMap.java:
```diff
@@ -1,26 +1,65 @@
+    public DifferentialDriveMap getDriveMap() {
+        return new DifferentialDriveMap() {
+
+<<<<<<< HEAD
```
This doesn't seem like it's in the right place...

msoucy commented on src/main/java/frc/robot/maps/TempestMap.java:
```diff
@@ -1,39 +1,89 @@
 package frc.robot.maps;
 
+<<<<<<< HEAD
```
Merge conflict

msoucy commented on src/main/java/frc/robot/maps/TempestMap.java:
```diff
@@ -1,39 +1,89 @@
-import edu.wpi.first.wpilibj.SpeedControllerGroup;
-import edu.wpi.first.wpilibj.Talon;
-
+<<<<<<< HEAD
```
Merge conflict

msoucy commented on src/main/java/frc/robot/maps/TempestMap.java:
```diff
@@ -1,39 +1,89 @@
-        return new DriveMap() {
+    public DifferentialDriveMap getDriveMap() {
+        return new DifferentialDriveMap() {
+<<<<<<< HEAD
```
Merge conflict

msoucy commented on src/main/java/frc/robot/maps/TempestMap.java:
```diff
@@ -1,39 +1,89 @@
+>>>>>>> Drive
             }
 
+<<<<<<< HEAD
```
Merge conflict

msoucy commented on src/main/java/frc/robot/maps/HyperionMap.java:
```diff
@@ -1,26 +1,32 @@
-    @Override
-    public DriveMap getDriveMap() {
-        return new DriveMap() {
+    public DifferentialDriveMap getDriveMap() {
```
This should be `@Override`

msoucy replied:
Does this chassis have a gyro?

msoucy commented on src/main/java/frc/robot/maps/TempestMap.java:
```diff
@@ -1,39 +1,32 @@
-            public SendableSpeedController left() {
-                return SendableSpeedController.wrap(new SpeedControllerGroup(new WPI_TalonSRX(4), new WPI_TalonSRX(1)));
-            }
+    public DifferentialDriveMap getDriveMap() {
```
Does this chassis have a gyro?
